homeassistant/components/russound_rio/entity.py

```diff
@@ -0,0 +1,73 @@
+"""Base entity for Russound RIO integration."""
+
+from collections.abc import Awaitable, Callable, Coroutine
+from functools import wraps
+from typing import Any, Concatenate, ParamSpec, TypeVar
+
+from aiorussound import Controller
+
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, RUSSOUND_RIO_EXCEPTIONS
+
+_P = ParamSpec("_P")
+_EntityT = TypeVar("_EntityT", bound="RussoundBaseEntity")
```

Comment:
Bound might not be necessary here, could just put in RussoundBaseEntity as a type in function directly. Don't know the python semantics regarding this.

Reply:
There are new typevars for this, I'm not sure about the usage of them. I think `technove` has one

Reply:
Yep, they are just using the shorthand. I'm swapping over to that.